### DISCUSSION TITLE
Support passing multiple i18next modules to RemixI18Next

### DISCUSSION THREAD

**Discussion by trishrempel** [sergiodxa/remix-i18next#136] [Ideas]
Currently, you can only pass an i18next back end module to RemixI18next using the `backend` option.

However, it would be useful to be able to pass one or more modules of any type that can be passed to the internal i18next instance.
In my particular use case, I'd like to configure RemixI18next to use a custom `I18nFormat` plugin.

### Proposal

Introduce an optional `plugins` option that is an array of i18next plugins.
With this, you could deprecate the `backend` option and have it passed in `modules` instead.

```js
export interface RemixI18NextOption {
  /**
   * The i18next options used to initialize the internal i18next instance.
   */
  i18next?: Omit<InitOptions, "react" | "detection"> | null;
  /**
   * @deprecated Use `plugins` instead.
   * The i18next backend module used to load the translations when creating a
   * new TFunction.
   */
  backend?: NewableModule<BackendModule<unknown>>;
  /**
   * The i18next plugins used to extend the internal i18next instance.
   */
  plugins?: NewableModule<Module>[];
  detection: LanguageDetectorOption;
}
```

We can then update `createInstance()` as follows:
```js
  private async createInstance(options: Omit<InitOptions, "react"> = {}) {
    let instance = createInstance();
    const plugins = [
      ...(this.options.backend ? [this.options.backend] : []),
      ...(this.options.plugins || []),
    ];
    for (const plugin of plugins) instance.use(plugin);
    await instance.init(options);
    return instance;
  }
```

**Comment by trishrempel** [comment]
Pull request to implement this: https://github.com/sergiodxa/remix-i18next/pull/137

**Comment by sergiodxa** [comment]
This is now published on the latest version of remix-i18next
